### PR TITLE
added code to generate fairprep result with output

### PR DESCRIPTION
**Pull request by vvthakral**

### CODE REVIEW

File: examples/fair_prep_results.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import os
+import sys
+import shutil
+import warnings
+warnings.simplefilter('ignore')
+
+from helper import extract_info
+
+sys.path.append(os.getcwd())
```
Comment (sscdotopen) [inline examples/fair_prep_results.py:9]: I thought we don't need this if we run the examples from the root folder?

File: examples/fair_prep_results.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import numpy as np
+import matplotlib.pyplot as plt
+
+#creating list of parameters that we will alter to observe variations
+seeds = [0xbeef, 0xcafe, 0xdead, 0xdeadcafe, 0xdeadbeef, 0xbeefcafe, 0xcafebeef, 50, 583, 5278, 100000, 0xefac,0xfeeb, 0xdaed, 0xefacdaed, 0xfeebdead]
+learners = [NonTunedLogisticRegression(), LogisticRegression(), NonTunedDecisionTree(), DecisionTree()]
+processors = [(NoPreProcessing(), NoPostProcessing()), (DIRemover(1.0), NoPostProcessing()), (DIRemover(0.5), NoPostProcessing()), (Reweighing(), NoPostProcessing()),
+              (NoPreProcessing(), RejectOptionPostProcessing()), (NoPreProcessing(), CalibratedEqualOddsPostProcessing())]
+
+def calc_metrics(seed, learn, pre, post):
```
Comment (sscdotopen) [inline examples/fair_prep_results.py:28]: please use long variable and methods names like calculate_metrics, pre_processors, post_processorrs

File: examples/fair_prep_results.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        post_processors=[post])
+    exp.run()
+
+
+def run_exp(accuracy, disp_imp, fnr, fpr, seeds, learners, processors):
+    '''
+    This is the main driver function that calls the calc_metrics to give metrices on combinations of various learners, pre and post processing techniques.
+    '''
+    for processor in processors:
+        for learn in learners:
```
Comment (sscdotopen) [inline examples/fair_prep_results.py:48]: learner

File: examples/fair_prep_results.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                print(learner_acc)
+            accuracy.append(learner_acc)
+            disp_imp.append(learner_di)
+            fnr.append(learner_fnr)
+            fpr.append(learner_fpr)
+    
+    return accuracy, disp_imp, fnr, fpr
+
+
+accuracy = []
```
Comment (sscdotopen) [inline examples/fair_prep_results.py:62]: why do we need to initialize these to empty lists?

File: examples/fair_prep_results.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        axs[loc].grid(True)
+        axs[loc].set_xlabel(x_label)
+        axs[loc].set_ylabel('Accuracy')
+        axs[loc].legend(label_list[int(i%4/2)])
+    fig.suptitle(main_title)
+    plt.subplots_adjust(wspace=0.3, hspace=0.43)
+    fig.savefig('examples/'+main_title+'.png')
+    plt.show()
+    
+plotter(x = disp_imp, y = accuracy, x_ticks = [0.5,1,1.5], x_label = 'DI', main_title = 'accuracy vs DI')
```
Comment (sscdotopen) [inline examples/fair_prep_results.py:92]: no spaces around = for named arguments please

File: examples/fair_prep_results.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+plotter(x = fnr, y = accuracy, x_ticks = [-0.4,0,0.4], x_label = 'FNR', main_title = 'accuracy vs FNR')
+plotter(x = fpr, y = accuracy, x_ticks = [-0.4,0,0.4], x_label = 'FPR', main_title = 'accuracy vs FPR')
```
Comment (sscdotopen) [inline examples/fair_prep_results.py:94]: please add a newline at the end of the file to be POSIX compliant

File: examples/helper.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
```
Comment (sscdotopen) [inline examples/helper.py:4]: please remove comments

File: examples/helper.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+                    fnr = (df['parameter'] == 'generalized_false_negative_rate') & test_data
+                    fnr_value = df.loc[fnr,:]
+                    learner_fnr.append(fnr_value.iloc[0,-1])
+
+                    fpr = (df['parameter'] == 'generalized_false_positive_rate') & test_data
+                    fpr_value = df.loc[fpr,:]
+                    learner_fpr.append(fpr_value.iloc[0,-1])
+
+                    shutil.rmtree(path+'/'+sub_folder)
```
Comment (sscdotopen) [inline examples/helper.py:41]: spaces around +

File: examples/fair_prep_results.py (hunk 1 above)
Comment (vvthakral) [reply examples/fair_prep_results.py:9]: We need it for running from the root directory. 
When we try to run a file in the examples folder from the root, sys.path stores the path of the examples directory but does not store the path of the root. So we need to add the path of root to the path list.

File: examples/fair_prep_results.py (hunk 2 above)
Comment (vvthakral) [reply examples/fair_prep_results.py:28]: Okay.

File: examples/fair_prep_results.py (hunk 3 above)
Comment (vvthakral) [reply examples/fair_prep_results.py:48]: Okay

File: examples/fair_prep_results.py (hunk 5 above)
Comment (vvthakral) [reply examples/fair_prep_results.py:92]: changed

File: examples/fair_prep_results.py (hunk 6 above)
Comment (vvthakral) [reply examples/fair_prep_results.py:94]: Added

File: examples/helper.py (hunk 7 above)
Comment (vvthakral) [reply examples/helper.py:4]: done

File: examples/helper.py (hunk 8 above)
Comment (vvthakral) [reply examples/helper.py:41]: added

**Comment by sscdotopen** [comment]
Please resolve the conflicts. You have to make sure your branch contains the latest changes from the master branch before you push it again.

**Pull request closed without merging**